File: exercises/intro-cpp/exercise04.cpp
// =================================================================
//
// File: exercise04.cpp
// Author(s): Esteban Padilla Cerdio - A01703068
//            Hilda Olivia Beltrán Acosta - A01251916
// Description: This file implements the PI approximation using the
//				series proposed by Euler.
//				pi = sqrt ( 6 * sumatoria(i = 1-N) (1 / i^2) )
//				The time this implementation takes will be used as
//				the basis to calculate the improvement obtained with
//				parallel technologies.
//
// Copyright (c) 2022 by Tecnologico de Monterrey.
// All Rights Reserved. May be reproduced for any non-commercial
// purpose.
//
// =================================================================

#include "utils.h"
#include <cmath>
#include <iomanip>
#include <iostream>

const int LIMIT = 1000000; // 1e6

using namespace std;

// implement your code here

int main(int argc, char *argv[])
{
	long double result;
	double ms;

	cout << "Starting..." << endl;
	ms = 0;
	for (int i = 0; i < N; i++)
	{
		start_timer();

		result = 0;

		// pi = sqrt ( 6 * sum(i = 1-N) (1 / i^2) )
		for (long long j = 1; j <= LIMIT; j++)
		{
			result += 1.0 / (j * j);
		}

		result = sqrt(6 * result);

		ms += stop_timer();
	}
<<<<<<< HEAD
	cout << "result = " << setprecision(10) << result << "\n";
	cout << "avg time = " << setprecision(10) << (ms / N) << " ms" << endl;
	== == == =
				 cout << "result = " << fixed << setprecision(10) << result << "\n";
	cout << "avg time = " << setprecision(5) << (ms / N) << " ms" << endl;
>>>>>>> 7518a1397c1d2682a3780a9d66fad3a64f4d9ff8

	return 0;
}

File: exercises/intro-python/exercise04.py
# =================================================================
##
## File: exercise04.py
# Author(s): Esteban Padilla Cerdio - A01703068
# 			 Hilda Olivia Beltrán Acosta - A01251916
# Description: This file implements the PI approximation using the
# series proposed by Euler.
##				pi = sqrt ( 6 * sumatoria(i = 1-N) (1 / i^2) )
# The time this implementation takes will be used as
# the basis to calculate the improvement obtained with
# parallel technologies.
##
# Copyright (c) 2022 by Tecnologico de Monterrey.
# All Rights Reserved. May be reproduced for any non-commercial
# purpose.
##
# =================================================================

import utils
import time


SIZE = 1000000  # 1e6


# Place your code here

if __name__ == "__main__":

    print("Starting...")
    startTime = endTime = ms = 0
    for i in range(utils.N):
        startTime = time.time() * 1000

        result = 0

        # pi = sqrt(6 * sum(i=1-N)(1 / i ^ 2))
        for j in range(1, LIMIT+1):
            result += 1.0 / (j * j)

        result = (6 * result)**(1/2)

        endTime = time.time() * 1000

        ms = ms + (endTime - startTime)

    print("result = ", result)
    print("avg time = ", (ms / utils.N), " ms")
